ouroboros-consensus/src/Ouroboros/Consensus/Util/CBOR/Simple.hs

```diff
@@ -0,0 +1,29 @@
+module Ouroboros.Consensus.Util.CBOR.Simple (
+    -- * Future-proofing
+    versionZeroProductFromCBOR
+  , versionZeroProductToCBOR
+  ) where
+
+import qualified Codec.CBOR.Decoding as CBOR
+import qualified Codec.CBOR.Encoding as CBOR
+import           Control.Monad (unless)
+
+{-------------------------------------------------------------------------------
+  Serialization aliases
+-------------------------------------------------------------------------------}
+
+-- | Simple codec to use since there's no version history to deal with and
+-- exactly one constructor
+versionZeroProductFromCBOR :: String -> Int -> CBOR.Decoder s a -> CBOR.Decoder s a
+versionZeroProductFromCBOR s arity dec = do
+    CBOR.decodeListLenOf (1 + arity)
+    tag <- CBOR.decodeTag
+    unless (0 == tag) $ fail s
+    dec
+
+-- | Simple codec to use since there's no version history to deal with and
+-- exactly one constructor
+versionZeroProductToCBOR :: [CBOR.Encoding] -> CBOR.Encoding
+versionZeroProductToCBOR xs =
```

Comment:
Suggestion: Like having a roundtrip property or something?

Reply:
This function is gone from this PR now